source/adapters/cuda/usm.cpp

```diff
@@ -36,10 +36,19 @@ urUSMHostAlloc(ur_context_handle_t hContext, const ur_usm_desc_t *pUSMDesc,
     return USMHostAllocImpl(ppMem, hContext, nullptr, size, alignment);
   }
 
-  auto UMFPool = hPool->HostMemPool.get();
-  *ppMem = umfPoolAlignedMalloc(UMFPool, size, alignment);
+  usm::pool_descriptor Desc = {hPool, hContext, nullptr, UR_USM_TYPE_HOST,
+                               false};
+  auto hPoolInternalOpt = hPool->PoolManager.getPool(Desc);
+  if (!hPoolInternalOpt.has_value()) {
+    // Internal error, every L0 context and usm pool should have Host, Device,
```

Comment:
```suggestion
    // Internal error, every CUDA context and usm pool should have Host, Device,
```

Should this say CUDA instead of L0? There's a few other comments below referencing L0 as well.

Reply:
Fixed.